coder_sniffer/Drupal/Sniffs/NamingConventions/ValidFunctionNameSniff.php

```diff
@@ -121,6 +128,25 @@ protected function processTokenOutsideScope(File $phpcsFile, $stackPtr)
             $phpcsFile->addError($error, $stackPtr, 'InvalidName', $data);
         }
 
+        $nameParts     = explode('.', basename($phpcsFile->getFilename()));
+        $fileExtension = end($nameParts);
+        // Validate function names only in *.module files.
+        if ($fileExtension === 'module') {
+            // Check if the function prefix is allowed to not respect standard.
+            foreach ($this->functionPrefixes as $functionPrefix) {
+                if (substr($functionName, 0, strlen($functionPrefix)) === $functionPrefix) {
+                    return;
+                }
+            }
+
+            $moduleName = reset($nameParts);
+            if (preg_match("/^_?$moduleName\_.+/", $functionName) === 0) {
+                $error = 'All functions defined in a module file must be prefixed with the module\'s name, found "%s".';
```

Comment:
We should give an example what we expect like "All functions defined in a module file must be prefixed with the module\'s name, found "%s" but expected "%s"". For the expected name we can just prefix the function name with `$moduleName . '_'`